perceval/converters/cqasm_converter.py

```diff
@@ -0,0 +1,331 @@
+# MIT License
+#
+# Copyright (c) 2024 Quandela
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# As a special exception, the copyright holders of exqalibur library give you
+# permission to combine exqalibur with code included in the standard release of
+# Perceval under the MIT license (or modified versions of such code). You may
+# copy and distribute such a combined system following the terms of the MIT
+# license for both exqalibur and Perceval. This exception for the usage of
+# exqalibur is limited to the python bindings used by Perceval.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from perceval.components import Processor, Source
+from perceval.components import Circuit, BS, PS, PERM
+from .abstract_converter import AGateConverter
+
+import cqasm.v3x as cqasm
```

Comment:
Are you referring to `import cqasm.v3x as cqasm` which is used throughout the class?

Reply:
Yes, please move the import to within the class. If it is imported at the module level - it becomes a required dependency of our package.

Reply:
I see how it is done for `qiskit` and `myqlm`, but I can't do something similar here because the `cqasm` package is used in 8 distinct methods - since the whole conversion process is broken down into several methods. My options seem to be:
1. re-import the package within every method call.
2. import at the class level and replace each occurrence of `cqasm` in the code by `CQASMConverter.cqasm.cqasm`.
3. Same as 2 but use `cqasm = CQASMConverter.cqasm` at the beginning of each method to make it look more compact.

Let me know what you prefer in terms of style.

Regarding making this library a dependency of the package, @ericbrts was ok to add `libqasm~=0.6.6` to `setup.py` and I went ahead with that. If it's not okay for you, then I'll remove it and use `try: import cqasm` at the top of the unit-test class.